sql/src/main/java/org/apache/druid/sql/calcite/run/NativeQueryMaker.java

```diff
@@ -87,7 +92,7 @@ public QueryResponse<Object[]> runQuery(final DruidQuery druidQuery)
 
     if (plannerContext.getPlannerConfig().isRequireTimeCondition()
         && !(druidQuery.getDataSource() instanceof InlineDataSource)) {
-      if (Intervals.ONLY_ETERNITY.equals(findBaseDataSourceIntervals(query))) {
+      if (!queryHasTimeFilter(query)) {
```

Comment:
can you move this over to `DataSourceAnalysis` ?
I feel that this would fit better there.

you could use `druidQuery.getDataSource().getAnalysis()` to access it here.

I think that should also handle the `druidQuery.getDataSource() instanceof InlineDataSource` correctly; and shouldn't be checked at this point.